### PR TITLE
feat: add rate limit docs

### LINKED ISSUE #925: Create Rate Limit docs📙
__Link to section:__
Please provide a specific link to the section in question. In the docs, you can click on a heading and then copy the relative link from the address bar.

__What is the issue?__
Please provide a detailed explanation of the problem, quoting parts of the docs or using screenshots if appropriate.

__Can you propose a solution?__
What changes do you think should be made? Have you considered multiple solutions? Will this be suitable for all use cases?

### PR DESCRIPTION
**Pull request by dblythy**
Closes #925

### CODE REVIEW

File: _includes/common/security.md [hunk 1]
```diff
@@ -562,6 +562,65 @@ Parse.Cloud.define("like", async request => {
 
 One very common use case for Cloud Code is sending push notifications to particular users. In general, clients can't be trusted to send push notifications directly, because they could modify the alert text, or push to people they shouldn't be able to. Your app's settings will allow you to set whether "client push" is enabled or not; we recommend that you make sure it's disabled. Instead, you should write Cloud Code functions that validate the data to be pushed and sent before sending a push.
 
+## Rate Limiting
+
+* Available only on parse-server starting 6.0.0 *
```
Comment (mtrezza) [inline _includes/common/security.md]:
```suggestion
* Available on Parse Server >=6.0.0 *
```

File: _includes/common/security.md [hunk 2]
```diff
@@ -562,6 +562,65 @@ Parse.Cloud.define("like", async request => {
...
+The valid options for a rate limit are:
+
+- `requestPath`: The path of the API route to be rate limited.
+- `requestMethods`: Optional, the HTTP request methods to be rate limited.
+- `requestTimeWindow`: The window of time in milliseconds within which the number of requests set in `requestCount` can be made before the rate limit is applied.
+- `requestCount`: The number of requests that can be made per IP address within the time window set in `requestTimeWindow` before the rate limit is applied.
+- `errorResponseMessage`: The error message that should be returned in the body of the HTTP 429 response when the rate limit is hit. Default is `Too many requests.`.
+- `includeInternalRequests`: Optional, whether the rate limit will also apply to requests that are made in by Cloud Code.
+- `includeMasterKey`: Optional, whether the rate limit will also apply to requests using the `masterKey`
+- `redisUrl` Optional, the URL of the Redis server to store rate limit data.
```
Comment (mtrezza) [inline _includes/common/security.md]:
```suggestion
It's important to restrict how often a client can call the Parse Server API. This prevents malicious attacks that could:
- overwhelm server resources by exceeding expected API traffic
- collect large amounts of data ("data scraping")
- repeatedly guess passwords, object IDs, installation IDs or other data ("brute force")

Parse Sever offers a mechanism to enforce rate limits by setting the Parse Server option `rateLimit`, or by specifying a `rateLimit` object on a Cloud Function validator.

The valid options for a rate limit are:

- `requestPath`: The path of the API route to be rate limited.
- `requestMethods`: Optional, the HTTP request methods to be rate limited.
- `requestTimeWindow`: The window of time in milliseconds within which the number of requests set in `requestCount` can be made before the rate limit is applied.
- `requestCount`: The number of requests that can be made per IP address within the time window set in `requestTimeWindow` before the rate limit is applied.
- `errorResponseMessage`: The error message that should be returned in the body of the HTTP 429 response when the rate limit is hit. Default is `Too many requests.`.
- `includeInternalRequests`: Optional, whether the rate limit will also apply to requests that are made in by Cloud Code.
- `includeMasterKey`: Optional, whether the rate limit will also apply to requests using the `masterKey`
- `redisUrl` Optional, the URL of the Redis server to store rate limit data.
```

File: _includes/common/security.md [hunk 3]
```diff
@@ -562,6 +562,65 @@ Parse.Cloud.define("like", async request => {
...
+Parse.Cloud.beforeSave('TestObject', () => {}, {
+  rateLimit: {
+    requestTimeWindow: 1 * 60 * 1000 // one write per minute,,
+    requestCount: 1,
+    errorResponseMessage: 'Too many requests!',
+  },
+});
+```
+
+
```
Comment (mtrezza) [inline _includes/common/security.md:625]:
```suggestion
> ⚠️ Rate limits should be enforced as far away from Parse Server as possible to mitigate possible impacts on resource costs, availability and integrity. While Parse Server offers a rate limiting mechanism as a conveniently available security feature without requiring a deep level of expertise, it is *not considered best practice* to enforce rate limits only after requests already reached the server. For better protection we advice to examine your network architecture an consider enforcing rate limits on the outer edge of the cloud if using a content delivery network, or at least before requests reach the server resource. Consult your cloud service provider for recommended rate limit and firewall solutions for your resources.

```

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Review by mtrezza — APPROVED** [review]
Looks good! Is this ready for merge?

**Comment by dblythy** [comment]
Yep!

**Pull request merged**